.github/workflows/inspektor-gadget.yml

```diff
@@ -1229,11 +1228,12 @@ jobs:
     # to secrets.
     # Also, unless specified in the secrets configuration, dependabot cannot access
     # to secrets.
-    # So, we skip this job when run by dependabot and from a PR coming from a
-    # fork.
+    # So, this job will be skipped when run by dependabot (if no secrets configured) and
+    # from a PR coming from a fork.
     # We still need to run this job on push, otherwise it will not be run when
-    # we merge something on main.
-    if: github.event_name == 'push' || (github.event.pull_request.head.repo.full_name == github.repository && github.actor != 'dependabot[bot]') && needs.check-secrets.outputs.cosign == 'true'
+    # we merge something on main. Also, we ensure this job will always run on
+    # for 'inspektor-gadget/inspektor-gadget' respository
+    if: needs.check-secrets.outputs.cosign == 'true' || (github.event_name == 'push' && github.repository == 'inspektor-gadget/inspektor-gadget')
```

Comment:
I should have explained the approach bit better :see_no_evil: 
> The comment refers to dependabot, but it was removed from the if, I suspect it will fail when run by the bot.

Actually the first check `needs.check-secrets.outputs.cosign == 'true'` ensures that the job will only run if secrets are configured so if no secrets are configured the CI won't run! 

> Moreover, this totally blocks forks from signing their image without tweaking the CI.

Do you mean they need to modify `(github.event_name == 'push' && github.repository == 'inspektor-gadget/inspektor-gadget')` ? 

I added this explicit check for our main repository so if somehow we delete cosign secrets the job will start to fail but for forks this check is ignored. I just [tried it on a fork ](https://github.com/mqasimsarfraz/inspektor-gadget/actions/runs/9131341274/job/25110631396) and configured the secrets and it seem to be working fine.

Reply:
I just did a test commit in fork to see if the job runs when secret is configured (only Action Secrets not Dependabot secrets): https://github.com/mqasimsarfraz/inspektor-gadget/commit/897d0407b0ce238a4478d55ec6b65f68a644532d and also another run when I remove them: https://github.com/mqasimsarfraz/inspektor-gadget/actions/runs/9595327823/job/26459837791 and it seem to be working as expected. 

Also, since we don't configure Dependabot secrets then job will be skipped for it.

Reply:
> You only shared the job using your public key, not the one without.
Can you please share the second?

Oops sorry. This is the link to the job without secrets: https://github.com/mqasimsarfraz/inspektor-gadget/actions/runs/9594669084